docs/source/building-supergraphs/router.mdx

```diff
@@ -345,53 +365,56 @@ class DataSourceWithServerId extends RemoteGraphQLDataSource {
-  supergraphSdl,
-  // highlight-start
-  buildService({ url }) {
-    return new DataSourceWithServerId({ url });
-  }
-  // highlight-end
-});
-
-const server = new ApolloServer({
-  gateway,
-  context() {
-    return { serverIds: [] };
-  },
-  plugins: [
+async function startApolloGateway() {
+  const gateway = new ApolloGateway({
+    supergraphSdl,
     // highlight-start
-    {
-      requestDidStart() {
-        return {
-          willSendResponse({ context, response }) {
-            // Append our final result to the outgoing response headers
-            response.http.headers.set(
-              'Server-Id',
-              context.serverIds.join(',')
-            );
-          }
-        };
-      }
+    buildService({ url }) {
+      return new DataSourceWithServerId({ url });
     }
     // highlight-end
-  ]
-});
+  });
+
+  const server = new ApolloServer({
+    gateway,
+    plugins: [
+      // highlight-start
+      {
+        requestDidStart() {
+          return {
+            willSendResponse({ contextValue, response }) {
+              // Append our final result to the outgoing response headers
+              response.http.headers.set(
+                'Server-Id',
+                contextValue.serverIds.join(',')
+              );
+            }
+          };
+        }
+      }
+      // highlight-end
+    ]
+  });
+  
+  const { url } = await startStandaloneServer(server, {
+    context() {
+      return { serverIds: [] };
+    },
+  });
 
-server.listen().then(({ url }) => {
-  console.log(`🚀 Server ready at ${url}`);
-});
+  console.log(`🚀  Server ready at ${url}`);
+}
 ```
 
 </ExpansionPanel>
 
-> In this example, multiple calls to `didReceiveResponse` are `push`ing a value onto the shared `context.serverIds` array. The order of these calls cannot be guaranteed. If you write logic that modifies the shared `context` object, make sure that modifications are not destructive, and that the order of modifications doesn't matter.
+> In this example, multiple calls to `didReceiveResponse` are `push`ing a value onto the shared `contextValue.serverIds` array. The order of these calls cannot be guaranteed. If you write logic that modifies the shared `contextValue` object, make sure that modifications are not destructive, and that the order of modifications doesn't matter.
 
 To learn more about `buildService` and `RemoteGraphQLDataSource`, see the [API docs](../api/apollo-gateway/).
 
 ## Custom directive support
 
-The `@apollo/gateway` library supports the use of custom [directives](/apollo-server/schema/directives/) in your subgraph schemas. This support differs depending on whether a given directive is a **type system directive** or an **executable directive**.
+The `@apollo/gateway` library supports the use of custom [directives](/apollo-server/v3/schema/directives/) in your subgraph schemas. This support differs depending on whether a given directive is a **type system directive** or an **executable directive**.
```

Comment:
Any reason for the v3 here?

This link still works:
https://www.apollographql.com/docs/apollo-server/schema/directives

Reply:
Oh my bad! 🤦‍♀️ I thought this linked to the old custom directives article, changing the link back!